File: src/schemas/rooms.py
from pydantic import BaseModel, Field, ConfigDict


class RoomAddRequest(BaseModel):
    title: str
    description: str | None = Field(None)
    price: int
    quantity: int

    allowed_species: list[str] | None = None
    temp_min: float | None = None
    temp_max: float | None = None
    humidity_min: float | None = None
    humidity_max: float | None = None
    room_conditions: str | None = None

    vaccinations_required: list[str] | None = None
    chip_required: bool = False

    diet_supported: list[str] | None = None
    feedings_per_day_max: int | None = None

    license_required: bool = False
    cohabitation_allowed: bool = True


class RoomAdd(RoomAddRequest):
    hotel_id: int


class Room(RoomAdd):
    id: int
    model_config = ConfigDict(from_attributes=True)


class PatchRoomRequest(BaseModel):
    title: str | None = None
    description: str | None = None
    price: int | None = None
    quantity: int | None = None

    allowed_species: list[str] | None = None
    temp_min: float | None = None
    temp_max: float | None = None
    humidity_min: float | None = None
    humidity_max: float | None = None
    room_conditions: str | None = None

    vaccinations_required: list[str] | None = None
    chip_required: bool | None = None

    diet_supported: list[str] | None = None
    feedings_per_day_max: int | None = None

    license_required: bool | None = None
    cohabitation_allowed: bool | None = None


class PatchRoom(PatchRoomRequest):
    hotel_id: int | None = None

File: hotels-frontend/src/api/rooms.ts
import { api } from "./client";

export type Room = {
  id: number;
  hotel_id: number;

  title: string;
  description?: string | null;

  price: number;
  quantity: number;
  available?: number;

  allowed_species?: string[] | null;
  temp_min?: number | null;
  temp_max?: number | null;
  humidity_min?: number | null;
  humidity_max?: number | null;

  room_conditions?: string | null;
  vaccinations_required?: string[] | null;
  chip_required?: boolean;

  diet_supported?: string[] | null;
  feedings_per_day_max?: number | null;

  license_required?: boolean;
  cohabitation_allowed?: boolean;
};

export type RoomCreate = {
  title: string;
  description?: string | null;

  price: number;
  quantity: number;

  allowed_species?: string[] | null;
  temp_min?: number | null;
  temp_max?: number | null;
  humidity_min?: number | null;
  humidity_max?: number | null;

  room_conditions?: string | null;
  vaccinations_required?: string[] | null;
  chip_required?: boolean;

  diet_supported?: string[] | null;
  feedings_per_day_max?: number | null;

  license_required?: boolean;
  cohabitation_allowed?: boolean;
};

export type RoomPatch = Partial<RoomCreate>;

export async function getRooms(
  hotelId: number,
  params?: { date_from?: string; date_to?: string }
) {
  const res = await api.get<Room[]>(`/rooms/${hotelId}/rooms`, { params });
  return res.data;
}

export async function getRoomsByHotel(
  hotelId: number,
  params?: { date_from?: string; date_to?: string }
) {
  const res = await api.get<Room[]>(`/rooms/${hotelId}/rooms`, { params });
  return res.data;
}

export async function createRoom(hotelId: number, payload: RoomCreate) {
  const res = await api.post(`/rooms/${hotelId}/rooms`, payload);
  return res.data;
}

export async function deleteRoom(hotelId: number, roomId: number) {
  const res = await api.delete(`/rooms/${hotelId}/rooms/${roomId}`);
  return res.data;
}

export async function updateRoom(
  hotelId: number,
  roomId: number,
  payload: RoomPatch
) {
  const res = await api.patch(`/rooms/${hotelId}/rooms/${roomId}`, payload);
  return res.data;
}